platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/serialize/ClassLoading.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.cc.impl.serialize
+
+
+private
+val defaultClassLoader = DefaultClassEncoder::class.java.classLoader
+
+
+internal
+fun classForName(name: String, classLoader: ClassLoader?): Class<*> =
+    try {
+        Class.forName(name, false, classLoader ?: defaultClassLoader)
+    } catch (e: ClassNotFoundException) {
+        throw ClassNotFoundException("Class '$name' not found in ${describeOptionalClassLoader(classLoader)}.", e)
```

Comment:
❓ 

Do we expect anything interesting from the CNFE exception stacktrace? Otherwise, at best, set `e` as suppressed exception, instead of a cause (or just discard it)?

Reply:
Well, it could report a different class than the one being loaded? Aren't those `NoClassDefFoundErrors`?